### PR TITLE
kb(grid):added kb on how to pre-set checkbox filter on initial load

### PR DESCRIPTION
**Pull request by xristianstefanov**
- Added KB on how to pre-set checkbox filter on initial load of the Grid

### CODE REVIEW

File: knowledge-base/grid-pre-select-checkbox-filter.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: How can I pre-set a checkbox filter using a filter descriptor?
```
Comment (dimodi) [inline knowledge-base/grid-pre-select-checkbox-filter.md:2]: "can i" is unlikely to be used in a Google search.

```suggestion
title: How to pre-set a checkbox filter with a filter descriptor?
```

File: knowledge-base/grid-pre-select-checkbox-filter.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+description: 
+type: how-to
+page_title: How can I pre-set a checkbox filter using a filter descriptor?
```
Comment (dimodi) [inline knowledge-base/grid-pre-select-checkbox-filter.md:5]:
```suggestion
page_title: How to pre-set a checkbox list filter using a filter descriptor?
```

File: knowledge-base/grid-pre-select-checkbox-filter.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+How can I pre-set a checkbox filter using a filter descriptor? How to set checkbox filter on Grid first load? How can I statically specify the checkbox list content? How to set checkbox filter on the first initialization of the Grid?
+
+## Solution
+To pre-select checkbox filter in Grid using filter descriptor:
```
Comment (dimodi) [inline knowledge-base/grid-pre-select-checkbox-filter.md:28]:
```suggestion
To pre-select checkbox filter in the Grid using a filter descriptor:
```

File: knowledge-base/grid-pre-select-checkbox-filter.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+1. Use the [OnStateInit](https://docs.telerik.com/blazor-ui/components/grid/state#events) event of the Grid.
+2. Implement **CompositeFilterDescriptor** inside the OnStateInit handler.
```
Comment (dimodi) [inline knowledge-base/grid-pre-select-checkbox-filter.md:31]:
```suggestion
2. Implement a **CompositeFilterDescriptor** inside the OnStateInit handler.
```

File: knowledge-base/grid-pre-select-checkbox-filter.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+			<td>Grid for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+How can I pre-set a checkbox filter using a filter descriptor? How to set checkbox filter on Grid first load? How can I statically specify the checkbox list content? How to set checkbox filter on the first initialization of the Grid?
+
+## Solution
```
Comment (dimodi) [inline knowledge-base/grid-pre-select-checkbox-filter.md:27]: The solution is missing several steps - 
* creating a new GridState instance
* adding the new CompositeFilterDescriptor instance to the `FilterDescriptors` collection in the state
* applying the new state to the current Grid state

**Review by dimodi — CHANGES_REQUESTED** [review]
Make the step-by-step instructions more detailed. They need to a bit more sense without looking at the code below.

File: knowledge-base/grid-pre-select-checkbox-filter.md (hunk 5 above)
Comment (xristianstefanov) [reply knowledge-base/grid-pre-select-checkbox-filter.md:27]: Done.

**Review by dimodi — APPROVED** [review]

**Pull request merged**